nptdms/tdms_segment.py

```diff
@@ -569,17 +569,22 @@ def read_raw_data_index(self, f, raw_data_index_header, endianness):
             self.data_type.__name__, self.number_values)
 
         if (self.data_type.size is None and
-                self.data_type != types.String):
+                self.data_type not in (types.String, types.Void)):
             raise ValueError(
                 "Unsupported data type: %r" % self.data_type)
 
+        if self.data_type == types.Void and self.number_values != 0:
+            raise ValueError("Void data only supported for empty segments")
+
         # In TDMS version 2.0, 1 is the only valid value for dimension
         if dimension != 1:
             raise ValueError("Data dimension is not 1")
 
         # Variable length data types have total size
         if self.data_type == types.String:
             self.data_size = types.Uint64.read(f, endianness)
+        elif self.number_values == 0:
+            return 0
```

Comment:
This change looks odd as this method doesn't return from anywhere else. Are these lines needed? Is it ok to just use the else clause below which will set `self.data_size` to 0? It looks like it's initialized to 0 already so that should be fine

Reply:
Correct, good catch. Fixed in e53f64a3f49d6df56a80d56445a14ed7a285c580.

I also realized I had to switch `np.dtype('void')` in most places to `np.dtype('V8')`, as when you actually instantiate an array with `dtype('void')`, the actual type of the array is the concrete `V8` type.